mw_basic_auth.go

```diff
@@ -117,6 +111,66 @@ func (k *BasicAuthKeyIsValid) ProcessRequest(w http.ResponseWriter, r *http.Requ
 		ctxSetAuthToken(r, keyName)
 	}
 
-	// Request is valid, carry on
 	return nil, http.StatusOK
 }
+
+func (k *BasicAuthKeyIsValid) handleAuthFail(w http.ResponseWriter, r *http.Request, token string) (error, int) {
+
+	// Fire Authfailed Event
+	AuthFailed(k, r, token)
+
+	// Report in health check
+	reportHealthValue(k.Spec, KeyFailure, "-1")
+
+	return k.requestForBasicAuth(w, "User not authorised")
+}
+
+func (k *BasicAuthKeyIsValid) compareHashAndPassword(hash string, password string, logEntry *logrus.Entry) error {
+
```

Comment:
as a general comment - from where do these "JS-style" empty lines after func name come? (some lint?)

Reply:
It's been a habit that I've had for many years - I think it looks neater. Is not just js... PEP-8, PSR-2 etc.

Also for golang, because `{` is in-line with function, unlike k&r style, it looks clearer in my opinion.

Reply:
well I never seen that in Go code and it is not part of `gofmt`, from other point of view `gofmt` doesn't remove those empty lines